Feature: Documentation Translation (#250)
simde-utc/portail: documentation/frontend/redux.md

# Redux

Redux is used in order to manage data in a unique store.
All Redux's implementation is done in the `redux/` folder.

For more information on Redux, [have a look at this playlist](https://www.youtube.com/watch?v=1w-oQ-i1XB8&index=15&list=PLoYCgNOIyGABj2GQSlDRjgvXtqfDxKm5b).

## The 3 parts of Redux

### Action types

> First, we have to define **action types**. They are constant strings which unable to identify actions to perform in the different reducers. It is recommanded to define them in the `types.js` file. In order to define CRUD actions sets more easily, use [`createCrudTypes`](#Action-types-creaters).

Reviewer: ```suggestion
First, we have to define **action types**. They are constant strings which enable to identify actions to perform in the different reducers. It is recommended to define them in the `types.js` file. In order to define CRUD actions sets more easily, use [`createCrudTypes`](#Action-types-creaters).
```

### Actions

> Actions are objects that are scattered in the reducers. they generally content in a `type` and in a `payload`. They are defined in the `actions.js` file.

Reviewer: ```suggestion
Actions are objects that are scattered in the reducers. They generally contain a `type` and a `payload`. They are defined in the `actions.js` file.
```

### Reducers

> Reducers are fonctions with the following prototype: `function(prevState, action)`.

Reviewer: ```suggestion
Reducers are functions with the following prototype: `function(prevState, action)`.
```

Do not directly modify `prevState`, make a copy of it. They are defined in the `reducers.js` file.

## CRUD creaters

As th portal's api mainly follows CRUD (Create Read Update Delete) design, most part of the Redux actions also follows it. That is why we have created creaters of action types, of actions and of CRUD reducers. These functions are in `react/utils.js`.

Ressources names are put in the singular by convention.

### Action types creaters

`createCrudTypes(name)` creates a set of types that follows the following scheme from a resources name `name` : 

This set of types is used by the following functions.

### Creater of actions

`createCrudActionSet(actionTypes, uri, overrides = {})` creates a set of CRUD actions from these parameters :
- `actionTypes` : a set of actions generated from the `createCrudTypes` function.
- `uri` : the url piece located after the domain name. It unables to access the right resource in the API. For example `assos` unables to access to `https://assos.utc.fr/api/v1/assos`
- `overrides` : an object unabling to replace and to add actions to the action set.

> ### Creater of reducer

Reviewer: ```suggestion
### Creator of reducer
```
